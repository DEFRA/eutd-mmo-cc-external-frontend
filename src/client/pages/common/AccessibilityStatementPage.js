import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getAllUserAttributes } from '../../actions/userAttributes.actions';
import { AccessibilityStatementEnglish } from './AccessibilityStatementEnglish';
import { AccessibilityStatementWelsh } from './AccessibilityStatementWelsh';

const AccessibilityStatementPage = (props) => {
  let userSelectedLanguage;
  const userAttributeLanguage = props.userAttributes.find((val) => val.name === 'language');
  
  if (userAttributeLanguage) {
    userSelectedLanguage = userAttributeLanguage.value;
  }

  return (
    userSelectedLanguage === 'cy_UK' 
      ? <AccessibilityStatementWelsh />
      : <AccessibilityStatementEnglish />
  );
};

const mapStateToProps = state => {
  return {
    userAttributes: state.userAttributes
  };
};

const loadData = store => {
  return store.dispatch(getAllUserAttributes());
};

export default {
  loadData,
  component: withRouter(connect(mapStateToProps,
    {
      getAllUserAttributes
    })(AccessibilityStatementPage))
};
