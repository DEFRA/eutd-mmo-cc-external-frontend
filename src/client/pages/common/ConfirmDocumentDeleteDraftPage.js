import React from 'react';
import {withRouter} from 'react-router';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import _ from 'lodash';

import {
  saveConfirmDocumentDelete,
  addConfirmDocumentDelete,
  dispatchApiCallFailed
} from '../../actions';
import {
  Main,
  BackLink,
  Header,
  GridRow,
  GridCol,
  MultiChoice
} from 'govuk-react';

import PageTitle from '../../components/PageTitle';
import Form from '../../components/elements/Form';
import HelpLink from '../../components/HelpLink';
import ContinueButton from '../../components/elements/ContinueButton';
import ErrorIsland from '../../components/elements/ErrorIsland';

import { scrollToErrorIsland, onHandleErrorClick} from '../utils';
import SelectRadio from '../../components/elements/SelectRadio';
import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import { withTranslation } from 'react-i18next';
import i18n from '../../../../src/i18n';
class ConfirmDocumentDeleteDraftPage extends React.Component {

  componentDidMount(){
    i18n.changeLanguage();
    this.forceUpdate();
  }

  onChange = e => {
    const confirmDocumentDelete = _.cloneDeep(
      this.props.confirmDocumentDelete || {}
    );
    confirmDocumentDelete.documentDelete = e.target.value;
    this.props.addConfirmDocumentDelete(confirmDocumentDelete);
  };

  onSubmit = async e => {
    e.preventDefault();

    try {
      const {path, journey, previousUri, nextUri} = this.props.route;
      const documentNumber = this.props.match.params['documentNumber'];
      const documentDelete = await this.props.saveConfirmDocumentDelete(path, nextUri, journey, documentNumber);
      const to = (documentDelete && documentDelete === 'Yes') ? nextUri : previousUri;

      this.props.history.push(to);
    } catch (error) {
      scrollToErrorIsland();
    }
  };

  render() {
    const {documentDeleteError, errors} = this.props.errors;
    const {confirmDocumentDelete = {}, route, match, t} = this.props;
    const {path, previousUri, nextUri, journey, journeyText} = route;
    const title = t(`${journey}DeleteCatchCertificateDeleteDraftJourneytext`, {journeyText: t(journeyText)})
    const h1Text =  t(`${journey}DeleteCatchCertificateDraftDeleteConfirmationText`, {journeyText: t(journeyText)})
    const currentUri = path.replace(':documentNumber', match.params.documentNumber);
    return (
      <Main>
        {errors && (
          <ErrorIsland
            errors={errors.map(err => ({
              message: t(err.text),
              key: err.targetName
            }))}
            onHandleErrorClick={onHandleErrorClick}
          />
        )}

        <PageTitle
          title={`${!_.isEmpty(errors) ? 'Error: ' : ''}${title} - ${t(this.props.route.title)}`}
        />
        <GridRow>
            <BackLink onClick={e => { e.preventDefault(); this.props.history.push(previousUri); }} href={previousUri}>
            {t('commonBackLinkBackButtonLabel')}
              </BackLink>
        </GridRow>

        <Form action='/orchestration/api/v1/confirm-document-delete' currentUrl={currentUri} onSubmit={this.onSubmit}>
          <GridRow>
            <GridCol columnTwoThirds>
              <GridRow>
                <Header level='1'>
                  <legend>{h1Text}</legend>
                </Header>
              </GridRow>
              <MultiChoice meta={{touched: t(documentDeleteError), error: t(documentDeleteError)}} id='documentDelete'>
                <fieldset>
                  <SelectRadio id="documentDeleteYes" checked={confirmDocumentDelete.documentDelete === 'Yes'} value='Yes' name='documentDelete' inline onChange={this.onChange}>{t('Yes')}</SelectRadio>
                  <SelectRadio id="documentDeleteNo" checked={confirmDocumentDelete.documentDelete === 'No'} value='No' name='documentDelete' inline onChange={this.onChange}>{t('No')}</SelectRadio>
                </fieldset>
              </MultiChoice>
            </GridCol>
          </GridRow>
          <GridRow>
            <GridCol columnTwoThirds>
              <ContinueButton type='submit' id='continue'>{t('commonContinueButtonSaveAndContinueButton')}</ContinueButton>
            </GridCol>
          </GridRow>
          <input type="hidden" name="journey" value={journey} />
          <input type="hidden" name="currentUri" value={currentUri} />
          <input type="hidden" name="nextUri" value={nextUri} />
          <input type="hidden" name="previousUri" value={previousUri} />
        </Form>
        <HelpLink journey={route.journey} />
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    errors: state.errors,
    confirmDocumentDelete: state.confirmDocumentDelete
  };
}

function loadData(store) {
  if (Object.keys(this.queryParams).length !== 0) {
    store.dispatch(dispatchApiCallFailed(JSON.parse(this.queryParams.error)));
  }
}

export const component = withRouter(
  connect(
    mapStateToProps,
    {
      addConfirmDocumentDelete,
      saveConfirmDocumentDelete
    }
  )(withTranslation()(ConfirmDocumentDeleteDraftPage))
);

ConfirmDocumentDeleteDraftPage.propTypes = {
  confirmDocumentDelete: PropTypes.object.isRequired,
  errors: PropTypes.object
};

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
