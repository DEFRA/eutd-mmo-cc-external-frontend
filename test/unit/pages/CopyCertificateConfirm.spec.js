import _ from 'lodash';
import { mount } from 'enzyme';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import {act} from 'react-dom/test-utils';

import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { component as CopyCertificateConfirmPage } from '../../../src/client/pages/common/CopyCertificateConfirm';
import copyCertificateConfirmPage from '../../../src/client/pages/common/CopyCertificateConfirm';
import {
  addCopyDocument,
  clearCopyDocument,
  clearErrors,
  submitCopyCertificate,
  unauthorisedCopyDocument,
  checkCopyCertificate
} from '../../../src/client/actions/copy-document.actions';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import {
  dispatchApiCallFailed
} from '../../../src/client/actions';

jest.mock('../../../src/client/actions/copy-document.actions');
jest.mock('../../../src/client/actions');

const mockBeforeEach = () => {
  addCopyDocument.mockReturnValue({ type: 'CONFIRM_COPY_DOCUMENT' });
  clearCopyDocument.mockReturnValue({ type: 'CLEAR_COPY_DOCUMENT' });
  clearErrors.mockReturnValue({ type: 'CLEAR_ERRORS '});
  submitCopyCertificate.mockReturnValue({
    type: 'CONFIRM_COPY_DOCUMENT', payload: {
      documentNumber: 'document1',
      copyDocumentAcknowledged: true,
      voidDocumentConfirm: true
    }
  });
  unauthorisedCopyDocument.mockReturnValue({ type: 'unauthorised_copy_document' });
  checkCopyCertificate.mockReturnValue({type: 'UNAUTHORISED_COPY_DOCUMENT'});
  dispatchApiCallFailed.mockReturnValue({type: 'CALL_FAILED'});
};

const mockStore = configureStore([thunk.withExtraArgument({
  orchestrationApi: {
    get: () => {
      return new Promise(res => {
        res({});
      });
    }
  }
})]);

let props = {
  onSubmit: jest.fn().mockReturnValue({
    copyDocumentAcknowledged: true,
    documentNumber: 'document1',
    journey: 'catch certificate'
  }),
  onChange: jest.fn().mockReturnValue({
    copyDocumentAcknowledged: true,
    documentNumber: 'document1',
    journey: 'catch certificate'
  }),
  match: {
    path: '/create-catch-certificate/:documentNumber/copy-this-catch-certificate',
    url: '/create-catch-certificate/GBR-2021-CC-D6FBF748C/copy-this-catch-certificate',
    isExact: true,
    params: {'documentNumber': 'GBR-2021-CC-D6FBF748C'}
  },
  route: {
    path: '/create-catch-certificate/:documentNumber/copy-this-catch-certificate',
    title: 'Create a UK catch certificate - GOV.UK',
    journey: 'catchCertificate',
    journeyText: 'catch certificate',
    previousUri: '/create-catch-certificate/catch-certificates',
    nextUri: '/create-catch-certificate/:documentNumber/add-your-reference',
    copyCertificateOptions: [
      {
        label: 'Copy all certificate data',
        value: 'copyAllCertificateData',
        name: 'copyDocument',
        id: 'copyAllCertificateData',
        hint: 'Enables you to quickly create a new draft catch certificate from a previously submitted document with similar data',
      },
      {
        label: 'Copy all certificate data EXCEPT the landings',
        value: 'copyExcludeLandings',
        id: 'copyExcludeLandings',
        name: 'copyDocument',
        hint: 'Enables you to quickly create a new draft catch certificate from a previously submitted document with similar data, but exclude the landings in the copy',
      }, {
        label: 'Copy all certificate data AND void the original',
        value: 'voidDocumentConfirm',
        id: 'voidDocumentConfirm',
        name: 'copyDocument',
        hint: 'Allows you correct errors in a submitted document by copying and replacing it',
      }
    ]
  },
  clearErrors: jest.fn()
};

let storeDefaults = {
  confirmCopyDocument: {
    copyDocumentAcknowledged: false,
    voidDocumentConfirm: false
  },
  errors: { errors: [] },
};

jest.spyOn(window, 'scrollTo')
  .mockImplementation(() => {});

let wrapper = mount(
  <Provider store={mockStore(storeDefaults)}>
    <MemoryRouter>
      <CopyCertificateConfirmPage {...props} />
    </MemoryRouter>
  </Provider>
);

describe('snapshots describe', () => {
  beforeEach(() => {
   clearErrors.mockReturnValue({ type: 'CLEAR_ERRORS '});
    window.scrollTo = jest.fn();
  });

  

  it('should load successfully', () => {
    expect(wrapper).toBeDefined();
  });

  it('should take a snapshot of the whole page', () => {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  })
})


describe('CopyCertificateConfirm', () => {

  beforeEach(() => {
    mockBeforeEach();

    window.scrollTo = jest.fn();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should load successfully', () => {
    expect(wrapper).toBeDefined();
  });



  it('should have continue button on the page with title as per specs', () => {
    expect(wrapper.find({type: 'submit', id: 'continue'}).exists()).toBeTruthy();
    expect(wrapper.find({type: 'submit', id: 'continue'}).at(0).text()).toBe('Create draft catch certificate');
  });

  

  it('should have checkbox on the page', () => {
    expect(wrapper.exists('input#copyDocumentAcknowledged')).toBeTruthy();
    expect(wrapper.find('Checkbox#copyDocumentAcknowledged').text()).toBe(
      'I understand I must not reuse the same catch certificate or landing data for catches that have already been exported'
    );
  });

  it('should have radio button options with corresponding labels and hints to copy document', () => {
    // Copy All Certificate Data Option
    expect(wrapper.find('input#copyAllCertificateData').exists()).toBeTruthy();
    expect(wrapper.find('input#copyAllCertificateData').props().checked).toBeFalsy();
    expect(wrapper.find('#copyAllCertificateData').contains('Copy all certificate data')).toBeTruthy();
    expect(wrapper.find('#copyAllCertificateData StyledHint').text()).toBe('Enables you to quickly create a new draft catch certificate from a previously submitted document with similar data');

    // Copy Exclude Landings Option
    expect(wrapper.find('input#copyExcludeLandings').exists()).toBeTruthy();
    expect(wrapper.find('input#copyExcludeLandings').props().checked).toBeFalsy();
    expect(wrapper.find('#copyExcludeLandings').contains('Copy all certificate data EXCEPT the landings')).toBeTruthy();
    expect(wrapper.find('#copyExcludeLandings StyledHint').text()).toBe('Enables you to quickly create a new draft catch certificate from a previously submitted document with similar data, but exclude the landings in the copy');

    // Copy Void Original Option
    expect(wrapper.find('input#voidDocumentConfirm').exists()).toBeTruthy();
    expect(wrapper.find('input#voidDocumentConfirm').props().checked).toBeFalsy();
    expect(wrapper.find('#voidDocumentConfirm').contains('Copy all certificate data AND void the original')).toBeTruthy();
    expect(wrapper.find('#voidDocumentConfirm StyledHint').text()).toBe('Allows you correct errors in a submitted document by copying and replacing it');
  });

  it('should have text of the warning with proper spaces as per specs', () => {
    expect(wrapper.find('#warningMessage').at(0).find('strong').text()).toBe('You must not use a catch certificate or landing data for catches that have already been exported. Knowingly reusing catch certificates or using landing data that relate to a previous export is a serious offence and may result in enforcement action being taken.');
  });

  it('should show warning text and acknowledgement heading in order as per specs', () => {
    expect(wrapper.find('Header').exists()).toBeTruthy();
    expect(wrapper.find('#warningMessageRow h2').text()).toBe('Acknowledgement');
    expect(wrapper.find('#warningMessageRow+#acknowledgementRow').exists()).toBeFalsy();
  });

  it('should show warning text', () => {
    expect(
      wrapper
        .find('div#warningMessage')
        .hasClass('warning-message')
    ).toBeTruthy();
  });

  it('should show acknowledgement head with large style', () => {
    expect(
      wrapper
        .find('#acknowledgementRow')
        .find('h2')
    ).toBeTruthy();
  });

  it('should call the onClick of the back link', () => {
    const mockPreventDefault = jest.fn();

    const baclLinkEl = wrapper.find('BackLink');
    baclLinkEl.simulate('click', {
      preventDefault: mockPreventDefault
    });

    expect(mockPreventDefault).toHaveBeenCalled();
  });


  it('should call the onClick of the Secondary Button', () => {
    const mockPreventDefault = jest.fn();

    const secondaryButtonEl = wrapper.find('button[name="cancel-copy-certificate-confirm-btn"]');
    secondaryButtonEl.simulate('click', {
      preventDefault: mockPreventDefault
    });

    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('should call the on change and set FormGroup - Copy All option selected', () => {
    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': null,
      'copyDocumentAcknowledged': false
    });
    act(() => {
      wrapper.find('Radio#copyAllCertificateData').props().onChange({
        target: {
          name: 'copyDocument',
          value: 'copyAllCertificateData',
          checked: true
        }
      });
    });

    expect(component.state()).toEqual({
      'copyDocument': 'copyAllCertificateData',
      'copyDocumentAcknowledged': false
    });
  });

  it('should call the on change and set FormGroup - Copy Excluding Landing option selected', () => {
    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': null,
      'copyDocumentAcknowledged': false
    });
    act(() => {
      wrapper.find('Radio#copyExcludeLandings').props().onChange({
        target: {
          name: 'copyDocument',
          value: 'copyExcludeLandings',
          checked: true
        }
      });
    });
    expect(component.state()).toEqual({
      'copyDocument': 'copyExcludeLandings',
      'copyDocumentAcknowledged': false
    });
  });

  it('should call the on change and set FormGroup - Void Document option selected', () => {
    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': null,
      'copyDocumentAcknowledged': false
    });
    act(() => {
      wrapper.find('Radio#voidDocumentConfirm').props().onChange({
        target: {
          name: 'copyDocument',
          value: 'voidDocumentConfirm',
          checked: true
        }
      });
    });
    expect(component.state()).toEqual({
      'copyDocument': 'voidDocumentConfirm',
      'copyDocumentAcknowledged': false
    });
  });

  it('should call the on change and set copyDocumentAcknowledge as true', () => {
    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': null,
      'copyDocumentAcknowledged': false
    });

    act(() => {
      wrapper.find('Checkbox#copyDocumentAcknowledged').props().onChange({ target: { name: 'copyDocumentAcknowledged', checked: true }});
    });
    expect(component.state()).toEqual({
      'copyDocument': null,
      'copyDocumentAcknowledged': true
    });
  });

  it('should test act onSubmit with FormGroup - copy all option selected and Acknowledgement TRUE', () => {

    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': 'copyAllCertificateData',
      'copyDocumentAcknowledged': true
    });

    const mockPreventDefault = jest.fn();

    const formEl = wrapper.find('form');
    act(() => formEl.props().onSubmit({
      preventDefault: mockPreventDefault
    }));

    expect(formEl.exists()).toBeTruthy();
    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('should test act onSubmit with FormGroup - Exclude Landings selected and Acknowledgement TRUE', () => {

    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': 'copyExcludeLandings',
      'copyDocumentAcknowledged': true
    });

    const mockPreventDefault = jest.fn();

    const formEl = wrapper.find('form');
    act(() => formEl.props().onSubmit({
      preventDefault: mockPreventDefault
    }));

    expect(formEl.exists()).toBeTruthy();
    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('should test act onSubmit with FormGroup - Void Original selected and Acknowledgement TRUE', () => {

    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': 'voidDocumentConfirm',
      'copyDocumentAcknowledged': true
    });

    const mockPreventDefault = jest.fn();

    const formEl = wrapper.find('form');
    act(() => formEl.props().onSubmit({
      preventDefault: mockPreventDefault
    }));

    expect(formEl.exists()).toBeTruthy();
    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('should test act onSubmit with FormGroup - copy all option selected with Acknowledgement FALSE', () => {

    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': 'copyAllCertificateData',
      'copyDocumentAcknowledged': true
    });

    const mockPreventDefault = jest.fn();

    const formEl = wrapper.find('form');
    act(() => formEl.props().onSubmit({
      preventDefault: mockPreventDefault
    }));

    expect(formEl.exists()).toBeTruthy();
    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('should test act onSubmit with FormGroup set to NULL with Acknowledgement FALSE', () => {

    const component = wrapper.find('copyCertificateConfirmPage');
    component.setState({
      'copyDocument': null,
      'copyDocumentAcknowledged': true
    });

    const mockPreventDefault = jest.fn();

    const formEl = wrapper.find('form');
    act(() => formEl.props().onSubmit({
      preventDefault: mockPreventDefault
    }));

    expect(formEl.exists()).toBeTruthy();
    expect(mockPreventDefault).toHaveBeenCalled();
  });

  it('should call onChange for Acknowledgement', () => {
    const onChangeFn = wrapper.find('copyCertificateConfirmPage').instance().props.onChange;
    const spyOnChangeFn = jest.spyOn(wrapper.find('copyCertificateConfirmPage').instance().props,'onChange');
    onChangeFn({target:{
      name:'copyDocumentAcknowledged',
      value: true,
      checked: true
    }
    });
    expect(spyOnChangeFn).toHaveBeenCalled();
  });

  it('should call onChange for copyAllCertificateData', () => {
    const onChangeFn = wrapper.find('copyCertificateConfirmPage').instance().props.onChange;
    const spyOnChangeFn = jest.spyOn(wrapper.find('copyCertificateConfirmPage').instance().props, 'onChange');
    onChangeFn({
      target: {
        name: 'copyAllCertificateData',
        value: 'copyAllCertificateData',
        checked: true
      }
    });
    expect(spyOnChangeFn).toHaveBeenCalled();
  });

  it('should call onChange for copyExcludeLandings', () => {
    const onChangeFn = wrapper.find('copyCertificateConfirmPage').instance().props.onChange;
    const spyOnChangeFn = jest.spyOn(wrapper.find('copyCertificateConfirmPage').instance().props, 'onChange');
    onChangeFn({
      target: {
        name: 'copyExcludeLandings',
        value: 'copyExcludeLandings',
        checked: true
      }
    });
    expect(spyOnChangeFn).toHaveBeenCalled();
  });

  it('should call onChange for voidDocumentConfirm', () => {
    const onChangeFn = wrapper.find('copyCertificateConfirmPage').instance().props.onChange;
    const spyOnChangeFn = jest.spyOn(wrapper.find('copyCertificateConfirmPage').instance().props, 'onChange');
    onChangeFn({
      target: {
        name: 'voidDocumentConfirm',
        value: 'voidDocumentConfirm',
        checked: true
      }
    });
    expect(spyOnChangeFn).toHaveBeenCalled();
  });

  it('should call onSubmit', () => {
    const submitFn = wrapper.find('copyCertificateConfirmPage').instance().props.onSubmit;
    const spyOnSubmit = jest.spyOn(wrapper.find('copyCertificateConfirmPage').instance().props,'onSubmit');
    submitFn();
    expect(spyOnSubmit).toHaveBeenCalled();
  });

  it('should handle on change events', () => {
    wrapper.find('Radio').first().find('input[type=\'radio\']').simulate('change', { target: { checked: true, name: 'my-radio', value: true }});
    expect(wrapper.find('copyCertificateConfirmPage').state()['my-radio']).toBeTruthy();
  });


  it('should show error if no option',() => {
    const store = _.cloneDeep(storeDefaults);

    store.errors = {
     errors: [{ 'targetName': 'copyDocumentAcknowledged', 'text': 'Check the acknowledgement to continue'},
               {'targetName': 'voidOriginal', 'text': 'Select an option to continue'}
              ]
    };

    wrapper = mount(
      <Provider store={mockStore(store)}>
        <MemoryRouter>
          <CopyCertificateConfirmPage {...props} />
        </MemoryRouter>
      </Provider>
    );

    let ErrorIsland = wrapper.find('div#errorIsland');
    expect(ErrorIsland.exists()).toBeTruthy();
    expect(wrapper.find('ul.error-summary-list li a[href="#voidOriginal"]').text()).toEqual('Select an option to continue');
    expect(wrapper.find('ul.error-summary-list li a[href="#copyDocumentAcknowledged"]').text()).toEqual('Check the acknowledgement to continue');
});



  it('should clear errors on component will unmount', () => {
    const mockDispatchClearErrors = jest.fn();

    new CopyCertificateConfirmPage.WrappedComponent({
      match: {
        params: {
          documentNumber: ''
        }
      },
      unauthorised: true,
      clearErrors: mockDispatchClearErrors
    }).componentWillUnmount();

    expect(mockDispatchClearErrors).toHaveBeenCalled();
  });

  it('should push history for component did mount', () => {
    const mockPush = jest.fn();

    new CopyCertificateConfirmPage.WrappedComponent({
      match: {
        params: {
          documentNumber: '',
        },
      },
      unauthorised: true,
      history: {
        push: mockPush,
      },
    }).componentDidMount();

    expect(mockPush).toHaveBeenCalled();
  });

  it('should push history for component did update', () => {
    const mockPush = jest.fn();

    new CopyCertificateConfirmPage.WrappedComponent({
      match: {
        params: {
          documentNumber: '',
        },
      },
      unauthorised: true,
      history: {
        push: mockPush,
      },
    }).componentDidUpdate();

    expect(mockPush).toHaveBeenCalled();
  });

  describe('loadData', () => { 
    const store = {
      dispatch: () => {
        return new Promise((res) => {
          res();
        });
      }
    };

    const documentNumber = 'some-document-number';

    it('will call all methods needed to load the component', () => {
      copyCertificateConfirmPage.documentNumber = documentNumber;

      copyCertificateConfirmPage.loadData(store);

      expect(unauthorisedCopyDocument).toHaveBeenCalled();
    });
  });


});


