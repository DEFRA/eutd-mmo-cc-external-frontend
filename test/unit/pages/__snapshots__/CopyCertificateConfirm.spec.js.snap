// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots describe should take a snapshot of the whole page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-n58wva-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/create-catch-certificate/catch-certificates"
          >
            Back
          </a>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/confirm-copy-certificate?c=/create-catch-certificate/undefined/copy-this-catch-certificate"
        method="POST"
      >
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h1
              class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
            >
              Copy undefined
            </h1>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="voidOriginal"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              
              <fieldset>
                <label
                  class="css-sdrcg6-StyledHoc"
                  for="copyAllCertificateData"
                  id="label-copyAllCertificateData"
                >
                  <input
                    class="css-1xt3i3l"
                    id="copyAllCertificateData"
                    name="copyDocument"
                    type="radio"
                    value="copyAllCertificateData"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Copy all certificate data
                  </span>
                  <span
                    class="css-hs8umk-StyledHint-StyledHoc e4u1u2g0"
                  >
                    Enables you to quickly create a new draft catch certificate from a previously submitted document with similar data
                  </span>
                </label>
                <label
                  class="css-sdrcg6-StyledHoc"
                  for="copyExcludeLandings"
                  id="label-copyExcludeLandings"
                >
                  <input
                    class="css-1xt3i3l"
                    id="copyExcludeLandings"
                    name="copyDocument"
                    type="radio"
                    value="copyExcludeLandings"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Copy all certificate data EXCEPT the landings
                  </span>
                  <span
                    class="css-hs8umk-StyledHint-StyledHoc e4u1u2g0"
                  >
                    Enables you to quickly create a new draft catch certificate from a previously submitted document with similar data, but exclude the landings in the copy
                  </span>
                </label>
                <label
                  class="css-sdrcg6-StyledHoc"
                  for="voidDocumentConfirm"
                  id="label-voidDocumentConfirm"
                >
                  <input
                    class="css-1xt3i3l"
                    id="voidDocumentConfirm"
                    name="copyDocument"
                    type="radio"
                    value="voidDocumentConfirm"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Copy all certificate data AND void the original
                  </span>
                  <span
                    class="css-hs8umk-StyledHint-StyledHoc e4u1u2g0"
                  >
                    Allows you correct errors in a submitted document by copying and replacing it
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
          id="warningMessageRow"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Acknowledgement
            </h2>
            <div
              class="warning-message e1vcssft0 css-jxvtpr-StyledWarningText-StyledHoc e4u1u2g0"
              id="warningMessage"
            >
              <div
                class="css-96kz36-IconImportantWrapper e4u1u2g1"
              >
                <svg
                  fill="currentColor"
                  height="100%"
                  preserveAspectRatio="xMidYMid meet"
                  style="display: block;"
                  version="1.1"
                  viewBox="0 0 35.000000 35.000000"
                  width="100%"
                >
                  <title>
                    icon important
                  </title>
                  <g
                    transform="translate(0.000000,35.000000) scale(0.100000,-0.100000)"
                  >
                    <path
                      d="M100 332 c-87 -48 -125 -155 -82 -232 48 -87 155 -125 232 -82 87 48
125 155 82 232 -48 87 -155 125 -232 82z m100 -122 c0 -53 -2 -60 -20 -60 -18
0 -20 7 -20 60 0 53 2 60 20 60 18 0 20 -7 20 -60z m0 -111 c0 -12 -7 -19 -20
-19 -19 0 -28 28 -14 43 11 11 34 -5 34 -24z"
                    />
                  </g>
                </svg>
              </div>
              <strong
                class="css-jegmsb-WarningTextWrapper e4u1u2g2"
              >
                You must not use a catch certificate or landing data for catches that have already been exported. Knowingly reusing catch certificates or using landing data that relate to a previous export is a serious offence and may result in enforcement action being taken.
              </strong>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="copyCertificate"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              
              <fieldset>
                <label
                  class="e1vcssft0 css-mpbcqr-StyledHoc"
                  for="copyDocumentAcknowledged"
                  id="label-copyDocumentAcknowledged"
                >
                  <input
                    class="css-17ieumk"
                    id="copyDocumentAcknowledged"
                    name="copyDocumentAcknowledged"
                    type="checkbox"
                    value="false"
                  />
                  <span
                    class="css-1ogj8nh"
                  >
                    I understand I must not reuse the same catch certificate or landing data for catches that have already been exported
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            id="cancel-copy-certificate-confirm-btn"
            name="cancel-copy-certificate-confirm-btn"
          >
            Cancel
            <span
              class="govuk-visually-hidden"
            >
              Cancel
            </span>
          </button>
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            name="continue"
            type="submit"
          >
            Create draft catch certificate
          </button>
        </div>
        <input
          name="journey"
          type="hidden"
          value="catchCertificate"
        />
        <input
          name="currentUri"
          type="hidden"
          value="/create-catch-certificate/undefined/copy-this-catch-certificate"
        />
        <input
          name="nextUri"
          type="hidden"
          value="/create-catch-certificate/:documentNumber/add-your-reference"
        />
        <input
          name="previousUri"
          type="hidden"
          value="/create-catch-certificate/catch-certificates"
        />
      </form>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
